csharp/Platform.Data.Doublets/Memory/United/Generic/UnitedMemoryLinksBase.cs

```diff
@@ -531,28 +533,28 @@ public virtual TLink Create(IList<TLink> restrictions)
                 freeLink = header.AllocatedLinks = Increment(header.AllocatedLinks);
                 _memory.UsedCapacity += LinkSizeInBytes;
             }
-            return freeLink;
+            return handler != null ? handler(Link<TLink>.Null, new Link<TLink>(freeLink, Constants.Null, Constants.Null)) : Constants.Continue;
         }
 
         /// <summary>
         /// <para>
-        /// Deletes the restrictions.
+        /// Deletes the substitution.
         /// </para>
         /// <para></para>
         /// </summary>
-        /// <param name="restrictions">
-        /// <para>The restrictions.</para>
+        /// <param name="restriction">
+        /// <para>The substitution.</para>
         /// <para></para>
         /// </param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public virtual TLink Delete(IList<TLink> restrictions)
+        public virtual TLink Delete(IList<TLink> restriction, WriteHandler<TLink> handler)
         {
             ref var header = ref GetHeaderReference();
-            var link = restrictions[Constants.IndexPart];
+            var link = restriction[Constants.IndexPart];
             if (LessThan(link, header.AllocatedLinks))
             {
                 UnusedLinksListMethods.AttachAsFirst(link);
-                return link;
+                return handler != null ? handler(GetLinkStruct(link), Link<TLink>.Null) : Constants.Continue;
```

Comment:
No, you should load before version of link before actual delete is started, not after it finished.